substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -47,6 +57,70 @@ pub mod versioned {
 	>;
 }
 
+mod v8 {
+	use super::*;
+
+	pub struct MigrateToV8<T>(sp_std::marker::PhantomData<T>);
+
+	impl<T: Config> MigrateToV8<T> {
+		fn calculate_tvu() -> BalanceOf<T> {
+			BondedPools::<T>::iter()
+				.map(|(id, inner)| {
+					let bonded_account = BondedPool { id, inner: inner.clone() }.bonded_account();
+					let total_stake = T::Staking::total_stake(&bonded_account).unwrap_or_default();
+					let active_stake =
+						T::Staking::active_stake(&bonded_account).unwrap_or_default();
+					total_stake - active_stake
```

Comment:
Should use safe math here I think

Reply:
`active_stake <= total_stake && active_stake >= 0` should always hold right? If that's the contract, I would rather let the migration fail than carry on with the wrong states in this case.

Reply:
It wouldn't fail, it would silently underflow :P